### PR TITLE
Update learning rate comments and add assertions for reference model parameters in GRPO and RLOO tests

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/test_grpo_trainer.py [hunk 1]
```diff
@@ -387,7 +387,7 @@ def test_training_peft_model(self):
 
         training_args = GRPOConfig(
             output_dir=self.tmp_dir,
-            learning_rate=0.1,  # increase the learning rate to speed up the test
+            learning_rate=0.1,  # use higher lr because gradients are tiny and default lr can stall updates
```
Comment (qgallouedec) [inline tests/test_grpo_trainer.py:390]: see https://github.com/huggingface/trl/pull/4810

File: tests/test_grpo_trainer.py [hunk 2]
```diff
@@ -842,6 +845,8 @@ def test_training_with_sync_ref_model(self):
         for n, param in previous_trainable_params.items():
             new_param = trainer.model.get_parameter(n)
             assert not torch.equal(param, new_param), f"Parameter {n} has not changed."
+            new_ref_param = trainer.ref_model.get_parameter(n)
+            assert not torch.equal(previous_ref_params[n], new_ref_param), f"Ref Parameter {n} has not changed."
```
Comment (albertvillanova) [inline tests/test_grpo_trainer.py:849]: Just a naive question: we are sure both models (model and ref_model) have the same parameters and parameter names?

Comment (albertvillanova) [reply tests/test_grpo_trainer.py:849]: Anyway, maybe we could make it more explicit with 2 loops like:
```python
# Check trainable params updated
for n, param in previous_trainable_params.items():
    new_param = trainer.model.get_parameter(n)
    assert not torch.equal(param, new_param), f"Parameter {n} has not changed."

# Check ref params updated
for n, param in previous_ref_params.items():
    new_ref_param = trainer.ref_model.get_parameter(n)
    assert not torch.equal(param, new_ref_param), f"Ref Parameter {n} has not changed."
```

File: tests/test_rloo_trainer.py
Comment (albertvillanova) [inline tests/test_rloo_trainer.py:629]: You explicitly added this to GRPO but not to RLOO:

```suggestion
            output_dir=self.tmp_dir,
            beta=0.1,  # ensure ref model is created so sync can update it
```

File: tests/test_rloo_trainer.py [hunk 3]
```diff
@@ -652,6 +654,8 @@ def test_training_with_sync_ref_model(self):
         for n, param in previous_trainable_params.items():
             new_param = trainer.model.get_parameter(n)
             assert not torch.equal(param, new_param), f"Parameter {n} has not changed."
+            new_ref_param = trainer.ref_model.get_parameter(n)
+            assert not torch.equal(previous_ref_params[n], new_ref_param), f"Ref Parameter {n} has not changed."
```
Comment (albertvillanova) [inline tests/test_rloo_trainer.py:659]: The same as above: maybe 2 loops?

**Review by albertvillanova — APPROVED** [review]
Thanks. Just some suggestions below.

File: tests/test_grpo_trainer.py (hunk 2 above)
Comment (qgallouedec) [reply tests/test_grpo_trainer.py:849]: > Just a naive question: we are sure both models (model and ref_model) have the same parameters and parameter names?

Yes, it's a hard requirement, it has to be the same model, only the weights are different.

File: tests/test_rloo_trainer.py
Comment (qgallouedec) [reply tests/test_rloo_trainer.py:629]: `beta>0` by default in RLOO, but you're right, it's probably better to have it explicitly.

File: tests/test_grpo_trainer.py (hunk 2 above)
Comment (qgallouedec) [reply tests/test_grpo_trainer.py:849]: Therefore, I prefer this single-loop option because it clearly indicates that the parameters match, and if for some reason they do not match (and magically, this doesn't make the training fail), the single loop will detect this case, unlike the double loop.
